plaso/engine/artifact_filters.py

```diff
@@ -75,71 +74,111 @@ def BuildFindSpecs(self, environment_variables=None):
+        artifact_list = find_specs_per_source_type[
+            artifact_types.TYPE_INDICATOR_FILE]
+        artifact_list.append(find_spec)
+        continue
+      logger.warning('Unknown find specification type: {0:s}'.format(
+          type(find_spec)))
 
     self._knowledge_base.SetValue(
-        self.KNOWLEDGE_BASE_VALUE, self._find_specs_per_source_type)
+        self.KNOWLEDGE_BASE_VALUE, find_specs_per_source_type)
+
+
+  def _BuildFindSpecsFromArtifact(self, definition, environment_variables):
+    """Builds find specifications from an artifact definition.
+
+    Args:
+      definition (artifacts.ArtifactDefinition): artifact definition.
+      environment_variables (list[EnvironmentVariableArtifact]):
+          environment variables.
+
+    Returns:
+      list[dfwinreg.FindSpec|dfvfs.FindSpec]: find specifications.
+    """
+    find_specs = []
+    for source in definition.sources:
+      if source.type_indicator == artifact_types.TYPE_INDICATOR_FILE:
+        for path_entry in set(source.paths):
+          specifications = self._BuildFindSpecsFromFileSource(
+              path_entry, source.separator, environment_variables,
+              self._knowledge_base.user_accounts)
+          find_specs.extend(specifications)
+
+      elif (source.type_indicator ==
+            artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
+        for key_path in set(source.keys):
+          if self.CheckKeyCompatibility(key_path):
+            specifications = self._BuildFindSpecsFromRegistrySource(key_path)
+            find_specs.extend(specifications)
+
+      elif (source.type_indicator ==
+            artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE):
+        # TODO: Handle Registry Values Once Supported in dfwinreg.
+        # https://github.com/log2timeline/dfwinreg/issues/98
+        logger.warning((
+            'Windows Registry values are not supported, extracting key: '
+            '"{0!s}"').format(source.key_value_pairs))
+
+        source_key_paths = {
+            key_value['key'] for key_value in source.key_value_pairs}
+        for key_path in source_key_paths:
+          if self.CheckKeyCompatibility(key_path):
+            specifications = self._BuildFindSpecsFromRegistrySource(key_path)
+            find_specs.extend(specifications)
+
+      elif (source.type_indicator ==
+            artifact_types.TYPE_INDICATOR_ARTIFACT_GROUP):
+        for name in source.names:
+          specifications = self._BuildFindSpecsFromGroupSource(
+              name, environment_variables)
+          find_specs.extend(specifications)
+
+      else:
+        logger.warning(
+            'Unsupported artifact definition source type: "{0:s}"'.format(
+                source.type_indicator))
+
+      return find_specs
+
+  def _BuildFindSpecsFromGroupSource(self, group_name, environment_variables):
+    """Builds find specifications from a artifact group source type.
+
+    Args:
+      group_name (str): artifact group name.
```

Comment:
This method appears to support more types of artifacts than a group, since `_BuildFindSpecsFromArtifact` does the heavy lifting. Is this intended? Would adding a check for the artifact type make sense? Or should we rename the method?

Reply:
The source type check is done in _BuildFindSpecsFromArtifact, I don't think it's worth making this method more generic. and changing the name. This is a private method, so we shouldn't have to worry about this being called from elsewhere.